src/graph/sampling/neighbor/neighbor.cc

```diff
@@ -18,13 +22,15 @@ using namespace dgl::aten;
 namespace dgl {
 namespace sampling {
 
-HeteroSubgraph ExcludeCertainEdges(
+std::pair<HeteroSubgraph, std::vector<FloatArray>> ExcludeCertainEdges(
     const HeteroSubgraph& sg,
-    const std::vector<IdArray>& exclude_edges) {
+    const std::vector<IdArray>& exclude_edges,
+    const std::vector<FloatArray> *weights = nullptr) {
```

Comment:
If exclusion is performed separately on edges and weights, then it will be slower. If I leave this function unchanged and write a version with weights, then there will be code duplication.

Reply:
Also, excluding edges after sampling is performed breaks the LABOR sampling algorithm. There ultimately needs to be a solution where the sampling is done from edges that can be included.

Reply:
There was a google doc about excluding edges etc. I proposed a solution with no overhead here: https://github.com/dmlc/dgl/issues/4441#issuecomment-1221634762

Reply:
Or one use weighted sampling for labor where one passes 0s for the edges that need to excluded. Weighted sampling should be faster than the existing cuda weighted neighbor sampling, didn't test them enough though.

Reply:
> If exclusion is performed separately on edges and weights, then it will be slower. If I leave this function unchanged and write a version with weights, then there will be code duplication.

Hmm I see.  You can leave this as a comment in the code then.